modules/storage-page-memory/src/main/java/org/apache/ignite/internal/storage/pagememory/index/sorted/io/SortedIndexTreeIo.java

```diff
@@ -129,14 +127,20 @@ default int compare(
 
         dataPageReader.traverse(link, indexColumnsTraversal, null);
 
-        ByteBuffer indexColumnsBuffer = ByteBuffer.wrap(indexColumnsTraversal.result());
+        ByteBuffer firstBinaryTupleBuffer = ByteBuffer.wrap(indexColumnsTraversal.result()).order(LITTLE_ENDIAN);
+        ByteBuffer secondBinaryTupleBuffer = rowKey.indexColumns().valueBuffer();
 
-        int cmp = binaryTupleComparator.compare(indexColumnsBuffer, sortedIndexRow.indexColumns().valueBuffer());
+        int cmp = binaryTupleComparator.compare(firstBinaryTupleBuffer, secondBinaryTupleBuffer);
 
-        if (cmp != 0) {
+        // Binary Tuple Prefixes don't have row IDs, so they can't be compared.
+        if (cmp != 0 || isPrefix(firstBinaryTupleBuffer) || isPrefix(secondBinaryTupleBuffer)) {
```

Comment:
`isPrefix(firstBinaryTupleBuffer)` can't be true, because you read it from data pages

Reply:
Fix it